### ISSUE TITLE
Resizing window : item-pane width becomes wrong

### ISSUE THREAD

**Issue by zuphilip** [zotero/zotero#604]
When decreasing the width of the window the panes are also resized. This works fine until the items-pane (middle pane) reaches its minimum value. Then the item-pane (right pane) will overflow the visible area because it seems that its width is wrongly fixed as well. A special case: if you use Win+Left Arrow in Windows to half the window on the left. First reported in #602 AFAIK with the picture (please note that the text in the item pane is cutted from the right):

![](https://cloud.githubusercontent.com/assets/3360192/5607486/f73ab8e2-942b-11e4-9741-de3bef98463b.png)

**Comment by dstillman** [comment]
I can't test on Windows at the moment, but I'm not seeing this on OS X (before or after d5b4c31575fa7d31d490accbdeb09bedb624d5b9). The sync button shifts all the over to the Locate button:

![min-width](https://cloud.githubusercontent.com/assets/783082/5694622/6996266e-9923-11e4-9f92-90b473da465a.png)

**Comment by zuphilip** [comment]
I can see this behaviour with and without the patch (Win7, Zotero for Firefox). Moreover, it is not only the toolbars and icons that are vanishing/cutted of from the right but also the item pane. When I am resizing the windows vertically I can see that the width of the collections-pane and item-pane is fixed and only the width of the items-pane is shrinking until it reaches the minimum. This corresponds to the fact that there is only for `#zotero-items-pane` the attribute `flex="1"`. By adding this flex attribute to the other panes seems to help...

**Comment by dstillman** [comment]
> Moreover, it is not only the toolbars and icons that are vanishing/cutted of from the right but also the item pane.

Right, but things getting cut off isn't fixable. When the Firefox window passes the minimum width, it will start to cut off the Zotero elements. (In Standalone, we define a minimum width on the window itself, so this doesn't happen.) Adding ```flex``` to ```#zotero-collections-pane``` and ```#zotero-item-pane``` just delays the cutting off by ensuring that both of those panes shrink to their minimum width before that happens, as opposed to just ```#zotero-items-pane```. That works well, as you say, but it does have the effect of causing the collections pane to jump larger when the item pane is collapsed. So we may need to manually fix that in code.

Not sure if there are other any undesirable consequences.

It'd probably be helpful to open this as a pull request, though, so that others can easily try this out and comment.

**Comment by mtd91429** [comment]
>Right, but things getting cut off isn't fixable. When the Firefox window passes the minimum width, it will start to cut off the Zotero elements.

Math will always win, I agree. However, I think that the real issue is that portions of the interface become inaccessible at these widths and the only way to view them is to restore the window to a larger dimension. Perhaps adding `overflow-x: auto` to `#zotero-pane` would sufficiently resolve this problem. At larger widths, the horizontal scroll-bar is hidden - but when the window is compressed, it appears

**Comment by zuphilip** [comment]
<blockquote>Right, but things getting cut off isn't fixable. When the Firefox window passes the minimum width, it will start to cut off the Zotero elements</blockquote>
Yeah, but this is only part of the problem. It starts when the items-pane is at minimal width (290px) independent of the window width. E.g. beginning with such a configuration in standalone on Windows:
![standalone-large](https://cloud.githubusercontent.com/assets/5199995/5696716/cd8536ea-99d9-11e4-8987-8b7156a75b9c.jpg)
...and decreasing the overall width:
![standalone-decreased](https://cloud.githubusercontent.com/assets/5199995/5696717/dcdb8cca-99d9-11e4-9258-bdffa8eae815.jpg)
...ends in a configuration where not all buttons and elements from the item-pane and toolbar are visible. However, the window width would still allow a good configuration of the pane width:
![standalone-large-better](https://cloud.githubusercontent.com/assets/5199995/5696729/707ada08-99da-11e4-9bef-98b7a32fe260.jpg)
@dstillman : Can you reproduce that on Mac?

I am beginning to <b>understand</b> how the resizing of the window is working and affecting the width of the zotero panes. I am not sure I can create or test any solution in the near future. @mtd91429 : `overflow-x` sounds also interesting for small width. Moreover, it would be possible to add some statements in the `resize` functions...

**Comment by dstillman** [comment]
> However, the window width would still allow a good configuration of the pane width

OK, I see what you mean. But again, the flex solution you suggested above seems mostly fine — it just has an undesirable side effect when collapsing the right-hand pane that I think we'd need to fix in code. I'm not going to work on this anytime soon, because the workaround is trivial, but I'm happy to look at a patch that also addressed that issue.

> Perhaps adding overflow-x: auto to #zotero-pane would sufficiently resolve this problem.

No, sorry, not going to do this.
